dateutil/rrule.py

```diff
@@ -1577,6 +1598,11 @@ def _parse_rfc(self, s,
                         from dateutil import parser
                     dtstart = parser.parse(value, ignoretz=ignoretz,
                                            tzinfos=tzinfos)
+                    if TZID is not None:
+                        if dtstart.tzinfo is None:
+                            dtstart = dtstart.replace(tzinfo=TZID)
+                        else:
+                            raise ValueError('DTSTART specifies multiple timezones')
```

Comment:
Did you happen to check of the RFC mentions whether or not this is allowed?

This is one of the two reasonable options - the other being that if `DTSTART` is timezone-aware (e.g. `20180204T0800Z`) and the rule has a specific time zone, you convert `DTSTART` into that time zone and generate dates starting from there. If the RFC is not silent on this point, we could choose either one, really, this one being more conservative and so probably preferable for a first release.